sklearn/model_selection/_split.py

```diff
@@ -1878,6 +1879,87 @@ def get_n_splits(self, X=None, y=None, groups=None):
         return len(self.unique_folds)
 
 
+class BinnedStratifiedKFold(StratifiedKFold):
+    """Stratified K-Folds cross-validator
+
+    Provides train/test indices to split data in train/test sets.
+
+    This cross-validation object is a variation of KFold that returns
+    stratified folds by binning a regression target. The folds are made
+    by approximately preserving the distribution of targets in each bin.
+
+    Read more in the :ref:`User Guide <cross_validation>`.
+
+    Parameters
+    ----------
+    n_splits : int, default=5
+        Number of folds. Must be at least 2.
+
+    shuffle : boolean, default=False
+        Whether to shuffle each stratification of the data before splitting
+        into batches.
+
+    n_bins : int, default=10
+        How many quantile bins to use.
+
+    random_state : None, int or RandomState
+        When ``shuffle=True``, pseudo-random number generator state used for
+        shuffling. If None, use default numpy RNG for shuffling.
```

Comment:
We do not need to explain random_state=None here. It's already included in the glossary.